spec/nmt.md

```diff
@@ -0,0 +1,303 @@
+
+**Leaf Nodes**: Each leaf in the tree represents the namespaced hash of a namespaced message `d = <NsID>||<Message Data>`. 
+The hash is computed as follows:
+
+```go
+NsH(d) = d.NsID || d.NsID || h(0x00, d)
+```
+The inclusion of the `0x00` value in the hash calculation serves as a leaf prefix and is done to conform to [RFC 6962](https://www.rfc-editor.org/rfc/rfc6962#section-2.1).
+
+**Non-leaf Nodes**: For an intermediary node `n` of the NMT with children
+
+`r` = `l.minNs || l.maxNs || l.hash` and
+
+`l` = `r.minNs || r.maxNs || r.hash`
+
+the namespaced hash is calculated as
+
+```go
+NsH(n) = min(l.minNs, r.minNs) || max(l.maxNs, r.maxNs) || h(0x01, l, r)
+```
+
+The inclusion of the `0x01` value in the hash calculation serves as a non-leaf prefix and is done to conform to [RFC 6962](https://www.rfc-editor.org/rfc/rfc6962#section-2.1).
+
+In an NMT data structure, the `minNs` and `maxNs` values of the root node denote the minimum and maximum namespace IDs, respectively, of all messages within the tree.
+
+
+An example of an NMT is shown in the figure below which utilizes SHA256 as the underlying hash function and namespace ID size of `1` byte. 
+The code snippets necessary to create this tree are provided in [NMT Library](#nmt-library) section, and the data items and tree nodes are represented as hex strings. 
+For the sake of brevity, we have only included the first 7 hex digits of SHA256 for each namespace hash.
+
+```markdown
+                                 00 03 b1c2cc5                                Tree Root
+                           /                       \
+                          /                         \
+                        NsH()                       NsH()  
+                        /                             \
+                       /                               \    
+               00 00 ead8d25                      01 03 52c7c03               Non-Leaf Nodes
+              /            \                    /               \
+            NsH()          NsH()              NsH()             NsH() 
+            /                \                /                   \
+    00 00 5fa0c9c       00 00 52385a0    01 01 71ca46a       03 03 b4a2792    Leaf Nodes
+        |                   |                 |                   |
+      NsH()               NsH()              NsH()               NsH()
+        |                   |                 |                   |
+00 6c6561665f30      00 6c6561665f31    01 6c6561665f32      03 6c6561665f33  Namespaced Data Items 
+
+        0                   1                  2                  3           Data Items/Leaves Indices
```

Comment:
Thanks for the suggestion. I initially considered using only the term `leaf indices`, but I realized that it could cause confusion about whether the row above is referring to leaves or data items. Therefore, I decided to include both `Data Items/Leaves Indices` to clarify the meaning.  What do you think?

Reply:
Since the row above has the label Namespaced Data Items, I got confused by seeing Data Items again on this row. I think Leaf Indices is clearer but defer to your judgement.